File: powerOfFour.js

/**
 * 
 * @param { integer } n 
 * @returns boolean if number is a power of four (4)
 */
const isPowerOfFour = (n) => {

    if(n === 4 || n === 1){
        return true
    }
    if(n % 4 !== 0 ){
        return false
    }

    return isPowerOfFour( Math.floor(n / 4) )

}

const solution = (input) => {
    let res = []
    for(let i in input){
        res[i] = isPowerOfFour(input[i])
    }

    return res
}


const input = [4, 2, 16, 9, 12, 64, 3, 27]
console.log(solution(input))

File: python/leetcode/power_of_four.py
class Solution(object):
    def isPowerOfFour(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == 0:
            return False
        if n == 1:
            return True
        
        if n > 1 and n < 4:
            return False
        
        if n % 4 != 0:
            return False
        
        return self.isPowerOfFour(n // 4)
        
        
input = 64
output = Solution().isPowerOfFour(input)
print(output)
            